examples/client.rs

```diff
@@ -179,15 +179,32 @@ fn main() {
         }
 
         let streams: Vec<u64> = conn.readable().collect();
-        for s in streams {
-            let data = conn.stream_recv(s, std::usize::MAX).unwrap();
+        'recv: for s in streams {
+            let (read, fin) = match conn.stream_recv(s, &mut buf) {
```

Comment:
I think it would be better to rewrite this whole thing as:
```rust
while let Ok((read, fin)) = conn.stream_recv(s, &mut buf) {
    debug!("{} received {} bytes", conn.trace_id(), read);

    let stream_buf = &buf[..read];
    ....
}
```
This way we read all of the data that is available, even if the output buffer is not big enough (and you get error handling wthout having to handle the different cases), and you wouldn't need to add the `'recv` label to the loop.